docs/large-tutorials/pyro.md

```diff
@@ -0,0 +1,1043 @@
+
+As before, we'll use the `Predictive` routine to generate samples. This time, though, we'll pass in the `guide`, which stores the variational distribution that is approximated to the posterior distribution. And, we'll start just by visualizing a subset of the parameters using the `return_sites` argument. 
+
+We can generate samples from the approximate posterior as follows
+
+```{code-cell} ipython3
+samples = Predictive(model, guide=guide, return_sites=['disk.incl', 'disk.Omega', 'disk.x_centroid', 'disk.y_centroid', 'disk.log_A_0', 'disk.log_sigma_0', 'disk.log_ring_amplitudes', 'disk.ring_means', 'disk.log_ring_sigmas'], num_samples=2000)(True)
+
+# extract samples from the Pyro Predictive object and convert units for convenience
+dict_samples = {k: np.expand_dims(v.detach().numpy(), 0) for k, v in samples.items()}
+
+# convert from radians to degrees
+for key in ["disk.incl", "disk.Omega"]:
+    dict_samples[key] /= deg
+    
+# convert from log values
+for key in ["disk.log_A_0", "disk.log_sigma_0", "disk.log_ring_amplitudes", "disk.log_ring_sigmas"]:
+    new_key = key.replace("log_", "")
+    dict_samples[new_key] = 10**dict_samples.pop(key)    
+```
+
+and then convert these samples to an ArviZ InferenceData object
+```{code-cell} ipython3
+import arviz as az
+dataset = az.convert_to_inference_data(dict_samples)
+dataset
+```
+
+Then, it is easy to use the ArviZ plotting routines to make many diagnostic plots. To start, let's visualize the 1D marginal posteriors
+
+
+```{code-cell} ipython3
+az.plot_posterior(dataset, var_names=["disk.Omega", "disk.incl", "disk.A_0", "disk.sigma_0"]);
+```
+
+And, we can also visualize the pairwise 2D marginal distributions (often called a "triangle" or "corner" plot)
+
+```{code-cell} ipython3
+az.plot_pair(dataset, var_names=["disk.ring_means"]);
+```
+
+As we mentioned, the lack of correlation between any parameters is *imposed* by the simple SVI guide that we used. This could be an issue if there were strong correlations between parameters. We'll address this limitiation in the next section by using a guide that incorporates correlations between parameters. 
+
+But first, let's see what the model and residuals look like for this optimized posterior distribution.
+
+```{code-cell} ipython3
+samples = Predictive(model, guide=guide, return_sites=['vis_real', 'vis_imag', 'sky_cube'], num_samples=1)(predictive=True)
+fig = compare_dirty_model_resid(samples["vis_real"][0], samples["vis_imag"][0], samples["sky_cube"][0]);
+```
+
+And the 1D profile
+
+```{code-cell} ipython3
+samples = Predictive(model, guide=guide, return_sites=['iprofile1D'], num_samples=50)(predictive=True)
+
+fig, ax = plt.subplots(nrows=1)
+
+for profile in samples["iprofile1D"]:
+    ax.plot(model.disk.R, profile, color="k", lw=0.2, alpha=0.2)
+    
+ax.set_xlabel("radius [au]")
+ax.set_ylabel(r"$I_\nu$ [Jy $\mathrm{arcsec}^{-2}$]");
+```
+
+## SVI with a MultiDiagonal Model
+
```

Comment:
A sentence or two here just to reiterate what's changing relative to the `AutoNormal` guide approach would be useful.